docs: add github codespace documentation

pikulet opened the pull request:
allows remote setup of this repository

[preview](https://github.com/pikulet/beyondmaf/blob/docs-codespace/docs/github-codespace.md)

@R9H9 you can try it out

DrSharky commented:
I'm not sure if I'd replace the current docs with this, more like have it as the primary, with the current ones as an alternative.

pikulet commented:
I added a line in the readme to link to this new document

And in the readme I added some sub-sections to make the setup more structured
- setup, node modules, env variables, start the site
